src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -117,21 +131,22 @@ public class RabbitMQMessageBus : MessageBusBase, IMessageBus {
                 return;
             _logger.Trace("Message Publish: {messageType}", messageType.FullName);
 
-            // RabbitMQ only supports delayed messages with a third party plugin.
-            if (delay.HasValue && delay.Value > TimeSpan.Zero) {
-                await AddDelayedMessageAsync(messageType, message, delay.Value).AnyContext();
-                return;
-            }
-
             var data = await _serializer.SerializeAsync(new MessageBusData {
                 Type = messageType.AssemblyQualifiedName,
                 Data = await _serializer.SerializeToStringAsync(message).AnyContext()
             }).AnyContext();
             
             var basicProperties = _publisherChannel.CreateBasicProperties();
             basicProperties.Persistent = _persistent;
-            basicProperties.Expiration = _defaultMessageTimeToLive.Milliseconds.ToString();
-
+            basicProperties.Expiration = _defaultMessageTimeToLive.TotalMilliseconds.ToString(CultureInfo.InvariantCulture);
+            // RabbitMQ only supports delayed messages with a third party plugin called "rabbitmq_delayed_message_exchange"
+            if (_delayedExchange && delay.HasValue && delay.Value > TimeSpan.Zero) {
```

Comment:
Yeah there is a problem if you try to create delayed and then actual exchange or vice versa with the *same name*.

Lets say you start with creating an exchange name as "foundatioExchange" and there is no plugin enabled for delay exchange. Now as per our logic we will always attempt to create x-delayed-type exchange and if that fails we fall back to the actual one. So now we have a successful creation of a regular exchange called "foundatioExchange".

Now you enable the plugin.

After enabling the plugin if you try to create a new RabbitMQMessageBus pub/sub with the same name as "foundatioExchange" you will be thrown with OperationInterruptedException .

+		o	{"The AMQP operation was interrupted: AMQP close-reason, initiated by Peer, code=406, text=\"PRECONDITION_FAILED - inequivalent arg 'type' for exchange 'FoundatioExchange' in vhost '/': received ''x-delayed-message'' but current is 'direct'\", classId=40, methodId=10, cause="}	RabbitMQ.Client.Exceptions.OperationInterruptedException


So you can't keep the same names for 2 different types of exchange. 
I have made some changes where we remove the extra boolean from the constructor and always create the delayed exchange first. If we fail then we fall back to the regular one. Currently testing various schenarios at this point. I will submit for new changes tomorrow.